### PR TITLE
[8.x] creating pivot tables by model classes

### PR DESCRIPTION
**Pull request by tyronehey**
This PR adds `createPivotTable` to the Schema builder class.

it creates `role_user` table with `role_id` and `user_id` columns by default. it also accepts a Closure as the third parameter that allows users to add their additional columns to the table.

### Usage 

```php
/**
 * Run the migrations.
 *
 * @return void
 */
public function up()
{
    Schema::createPivotTable('App\Models\User','App\Models\Role');
}
```

It's handy for users and allows them to create pivot tables quickly, and prevents them from writing similar codes to create tables like this.

### CODE REVIEW

File: src/Illuminate/Database/Schema/Builder.php [hunk 1]
```diff
@@ -200,6 +200,35 @@ public function create($table, Closure $callback)
         }));
     }
 
+    /**
+     * Create a new pivot table for a many-to-many relation on the schema.
+     *
+     * @param \Illuminate\Database\Eloquent\Model|string $firstModel
+     * @param \Illuminate\Database\Eloquent\Model|string $secondModel
+     * @param Closure|null $callback
```
Comment (driesvints) [inline src/Illuminate/Database/Schema/Builder.php:208]: Please make sure that Docblocks match the styling of the rest of the framework and use FQN for classes.

Comment (tyronehey) [reply src/Illuminate/Database/Schema/Builder.php]: Fixed it.

Comment (driesvints) [reply src/Illuminate/Database/Schema/Builder.php]: There should also be two spaces between the type and param name.

Comment (tyronehey) [reply src/Illuminate/Database/Schema/Builder.php]: There is two spaces between the type and param name in [631ad33](https://github.com/laravel/framework/pull/35397/commits/631ad332ae3c4b92ae0f4a287b4681f3078796d7)

Comment (driesvints) [reply src/Illuminate/Database/Schema/Builder.php]: There's still some StyleCI failures.

Comment (tyronehey) [reply src/Illuminate/Database/Schema/Builder.php]: Fixed it.

**Comment by taylorotwell** [comment]
Thanks for your pull request to Laravel!

Unfortunately, I'm going to delay merging this code for now. To preserve our ability to adequately maintain the framework, we need to be very careful regarding the amount of code we include. 

If possible, please consider releasing your code as a package so that the community can still take advantage of your contributions! 

If you feel absolutely certain that this code corrects a bug in the framework, please "**@**" mention me in a follow-up comment with further explanation so that GitHub will send me a notification of your response.

**Pull request closed without merging**